include/aws/common/logger.h

```diff
@@ -0,0 +1,102 @@
+#ifndef AWS_COMMON_LOGGER_H
+#define AWS_COMMON_LOGGER_H
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+enum aws_log_level
+{
+    aws_off = 0,
+    aws_fatal = 1,
+    aws_error = 2,
+    aws_warn = 3,
+    aws_info = 4,
+    aws_debug = 5,
+    aws_trace = 6
+};
+
+static inline const char *aws_get_log_level_name(enum aws_log_level level) {
+    switch(level) {
+        case aws_off:
+            return "OFF";
+        case aws_fatal:
+            return "FATAL";
+        case aws_error:
+            return "ERROR";
+        case aws_warn:
+            return "WARN";
+        case aws_info:
+            return "INFO";
+        case aws_debug:
+            return "DEBUG";
+        case aws_trace:
+            return "TRACE";
+        default:
+            return "UNKNOWN";
+    }
+}
+
+static inline void aws_log(enum aws_log_level level, struct aws_allocator *alloc, const char *tag, const char *msg,
```

Comment:
I think what we want is some macros that do nothing at the moment... something like

AWS_LOG_F(level, format, ...)
AWS_LOG(level, str)

And just have it do nothing for the moment.
This way it compiles out to nothing in the future when logging is disabled.

Reply:
I didn't want to use macros because:
1. I didn't want to conflict with the C++ SDK ones (couldn't come up with a better name quickly)
1. You can still get them to compile out to nothing if you define them with empty body.